### PR TITLE
DOCSUP-12152: Edit and translate MaterializeMySQL settings and check_sample_column_is_correct 

### PR DESCRIPTION
**Pull request by lehasm**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation (translation) for #26775

### CODE REVIEW

File: docs/en/engines/database-engines/materialized-mysql.md [hunk 1]
```diff
@@ -28,27 +26,31 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
+-   `max_bytes_in_buffer` — The maximum number of bytes allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_rows_in_buffers` — The maximum number of rows allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `65505`.
+-   `max_bytes_in_buffers` — The maximum number of bytes allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_flush_data_time` — The maximum number of seconds that data is allowed to be in cache for the whole database. When this number is exceeded, the data is materialized. Default: `1000`.
+-   `max_wait_time_when_mysql_unavailable` — A retry interval in milliseconds if MySQL is not available. A negative value disables retries. Default: `1000`.
+-   `allows_query_when_mysql_lost` — Whether to query materialized table if MySQL is not available. Default: `0` (`false`).
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-mysql.md:36]:
```suggestion
-   `allows_query_when_mysql_lost` — Indicates whether to query a materialized table if MySQL is not available. Default value: `0` (`false`).
```

File: docs/en/engines/database-engines/materialized-mysql.md [hunk 2]
```diff
@@ -28,27 +26,31 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
-        allows_query_when_mysql_lost=true,
-        max_wait_time_when_mysql_unavailable=10000;
-```
+
+-   `max_rows_in_buffer` — The maximum number of rows allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default: `65505`.
+-   `max_bytes_in_buffer` — The maximum number of bytes allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_rows_in_buffers` — The maximum number of rows allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `65505`.
+-   `max_bytes_in_buffers` — The maximum number of bytes allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_flush_data_time` — The maximum number of seconds that data is allowed to be in cache for the whole database. When this number is exceeded, the data is materialized. Default: `1000`.
+-   `max_wait_time_when_mysql_unavailable` — A retry interval in milliseconds if MySQL is not available. A negative value disables retries. Default: `1000`.
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-mysql.md:35]:
```suggestion
-   `max_rows_in_buffer` — The maximum number of rows allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default value: `65505`.
-   `max_bytes_in_buffer` — The maximum number of bytes allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default value: `1048576`.
-   `max_rows_in_buffers` — The maximum number of rows allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default value: `65505`.
-   `max_bytes_in_buffers` — The maximum number of bytes allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default value: `1048576`.
-   `max_flush_data_time` — The maximum number of seconds that data is allowed to be in cache for the whole database. When this number is exceeded, the data is materialized. Default value: `1000`.
-   `max_wait_time_when_mysql_unavailable` — A retry interval in milliseconds if MySQL is not available. A negative value disables retries. Default value: `1000`.
```

File: docs/en/engines/database-engines/materialized-mysql.md [hunk 3]
```diff
@@ -28,27 +26,31 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
 - `gtid_mode = on` since GTID based logging is a mandatory for providing correct `MaterializeMySQL` replication. Pay attention that while turning this mode `On` you should also specify `enforce_gtid_consistency = on`.
 
+**Example**
+
+```
+CREATE DATABASE mysql ENGINE = MaterializedMySQL('localhost:3306', 'db', 'user', '***')
+     SETTINGS
+        allows_query_when_mysql_lost=true,
+        max_wait_time_when_mysql_unavailable=10000;
+```
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-mysql.md:52]:
```suggestion
```sql
CREATE DATABASE mysql ENGINE = MaterializedMySQL('localhost:3306', 'db', 'user', '***')
     SETTINGS
        allows_query_when_mysql_lost=true,
        max_wait_time_when_mysql_unavailable=10000;
```
```

File: docs/en/engines/database-engines/materialized-mysql.md [hunk 4]
```diff
@@ -28,27 +26,31 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
 
-For the correct work of `MaterializeMySQL`, there are few mandatory `MySQL`-side configuration settings that should be set:
+For the correct work of `MaterializeMySQL`, there are a few mandatory `MySQL`-side configuration settings that should be set:
 
 - `default_authentication_plugin = mysql_native_password` since `MaterializeMySQL` can only authorize with this method.
 - `gtid_mode = on` since GTID based logging is a mandatory for providing correct `MaterializeMySQL` replication. Pay attention that while turning this mode `On` you should also specify `enforce_gtid_consistency = on`.
 
+**Example**
+
+```
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-mysql.md:47]:
```suggestion
```sql
```

File: docs/en/engines/database-engines/materialized-mysql.md [hunk 5]
```diff
@@ -28,27 +26,31 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
-     SETTINGS
-        allows_query_when_mysql_lost=true,
-        max_wait_time_when_mysql_unavailable=10000;
-```
+
+-   `max_rows_in_buffer` — The maximum number of rows allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default: `65505`.
+-   `max_bytes_in_buffer` — The maximum number of bytes allowed to be cached in memory for a single table. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_rows_in_buffers` — The maximum number of rows allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `65505`.
+-   `max_bytes_in_buffers` — The maximum number of bytes allowed to be cached in memory for the whole database. When this number is exceeded, the data is materialized. Default: `1048576`.
+-   `max_flush_data_time` — The maximum number of seconds that data is allowed to be in cache for the whole database. When this number is exceeded, the data is materialized. Default: `1000`.
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-mysql.md:34]:
```suggestion
-   `max_flush_data_time` — The maximum number of seconds that data is allowed to be in cache for the whole database. When this time is exceeded, the data is materialized. Default: `1000`.
```

File: docs/ru/engines/database-engines/materialized-mysql.md [hunk 6]
```diff
@@ -26,6 +24,32 @@ ENGINE = MaterializedMySQL('host:port', ['database' | database], 'user', 'passwo
...
+**Настройки на стороне сервера MySQL**
+
+Для правильной работы `MaterializeMySQL` на стороне сервера `MySQL` должны быть установлены обязательные настройки:
+
+- `default_authentication_plugin = mysql_native_password` - `MaterializeMySQL` может авторизоваться только с помощью такого метода.
+- `gtid_mode = on` - GTID логирование необходимо для работы репликации `MaterializeMySQL`. Обратите внимание, что при включении логирования необходимо также установить настройку `enforce_gtid_consistency = on`.
+
+**Пример**
+
+```
```
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-mysql.md:46]:
```suggestion
```sql
```

**Comment by tavplubix** [comment]
Is it still draft?

**Comment by lehasm** [comment]
This PR duplicates #27139

**Pull request closed without merging**